packages/ember-htmlbars/tests/integration/component_invocation_test.js

```diff
@@ -595,367 +66,14 @@ QUnit.test('moduleName is available on _renderNode when no layout is present', f
-    [OWNER]: owner,
-    layout: compile('{{#check-params id="expect-no"}}{{/check-params}}  {{#check-params id="expect-yes" as |foo|}}{{/check-params}}')
-  }).create();
-
-  runAppend(view);
-  equal(jQuery('#qunit-fixture #expect-no').text(), 'No');
-  equal(jQuery('#qunit-fixture #expect-yes').text(), 'Yes');
-});
-
-QUnit.test('non-expression hasBlockParams', function() {
-  owner.register('template:components/check-params', compile('{{#if hasBlockParams}}Yes{{else}}No{{/if}}'));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    layout: compile('{{#check-params id="expect-no"}}{{/check-params}}  {{#check-params id="expect-yes" as |foo|}}{{/check-params}}')
-  }).create();
-
-  runAppend(view);
-  equal(jQuery('#qunit-fixture #expect-no').text(), 'No');
-  equal(jQuery('#qunit-fixture #expect-yes').text(), 'Yes');
-});
-
-QUnit.test('components in template of a yielding component should have the proper parentView', function() {
-  var outer, innerTemplate, innerLayout;
-
-  owner.register('component:x-outer', Component.extend({
-    init() {
-      this._super(...arguments);
-      outer = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-template', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerTemplate = this;
-    }
-  }));
-
-  owner.register('component:x-inner-in-layout', Component.extend({
-    init() {
-      this._super(...arguments);
-      innerLayout = this;
-    }
-  }));
-
-  owner.register('template:components/x-outer', compile('{{x-inner-in-layout}}{{yield}}'));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    template: compile('{{#x-outer}}{{x-inner-in-template}}{{/x-outer}}')
-  }).create();
-
-  runAppend(view);
-
-  equal(innerTemplate.parentView, outer, 'receives the wrapping component as its parentView in template blocks');
-  equal(innerLayout.parentView, outer, 'receives the wrapping component as its parentView in layout');
-  equal(outer.parentView, view, 'x-outer receives the ambient scope as its parentView');
-});
-
-QUnit.test('newly-added sub-components get correct parentView', function() {
-  var outer, inner;
-
-  owner.register('component:x-outer', Component.extend({
-    init() {
-      this._super(...arguments);
-      outer = this;
-    }
-  }));
-
-  owner.register('component:x-inner', Component.extend({
-    init() {
-      this._super(...arguments);
-      inner = this;
-    }
-  }));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    template: compile('{{#x-outer}}{{#if view.showInner}}{{x-inner}}{{/if}}{{/x-outer}}'),
-    showInner: false
-  }).create();
-
-  runAppend(view);
-
-  run(() => { view.set('showInner', true); });
-
-  equal(inner.parentView, outer, 'receives the wrapping component as its parentView in template blocks');
-  equal(outer.parentView, view, 'x-outer receives the ambient scope as its parentView');
-});
-
-QUnit.test('components should receive the viewRegistry from the parent view', function() {
```

Comment:
:heavy_check_mark: components should receive the viewRegistry from the parent view